samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * Class managing Daily.co call functionalities including joining, leaving,
+ * handling audio/video tracks and UI updates for track states.
+ */
+class DailyCallManager {
+  constructor() {
+    this.call = DailyIframe.createCallObject();
+    this.isCameraMuted = null;
+    this.isMicMuted = null;
+    this.initialize();
+  }
+
+  /**
+   * Performs initial setup of event listeners and UI component interactions.
+   */
+  async initialize() {
+    await this.setupEventListeners();
+    document
+      .getElementById('toggle-camera')
+      .addEventListener('click', () => this.toggleCamera());
+    document
+      .getElementById('toggle-mic')
+      .addEventListener('click', () => this.toggleMicrophone());
+  }
+
+  /**
+   * Configures event listeners for various call-related events.
+   */
+  async setupEventListeners() {
+    const events = {
+      'active-speaker-change': this.handleActiveSpeakerChange.bind(this),
+      error: this.handleError.bind(this),
+      'joined-meeting': this.handleJoin.bind(this),
+      'participant-joined': this.handleJoin.bind(this),
+      'participant-left': this.handleJoin.bind(this),
+      'participant-updated': this.handleParticipantStateUpdate.bind(this),
+      'track-started': this.displayTrack.bind(this),
+      'track-stopped': this.destroyTrack.bind(this),
+    };
+
+    // Register the event handlers with the call object.
+    Object.entries(events).forEach(([event, handler]) => {
+      this.call.on(event, handler);
+    });
+  }
+
+  /**
+   * Handler for participant join events, updating participant count in the UI.
+   */
+  handleJoin() {
+    document.getElementById('participant-count').textContent = `Participants: ${
+      Object.keys(this.call.participants()).length
+    }`;
+    this.getTrackStates();
+  }
+
+  /**
+   * Handles errors emitted from the Daily call object.
+   * @param {Object} e - The error event object.
+   */
+  handleError(e) {
+    console.error('DAILY SENT AN ERROR!', e);
+    if (e.error?.details?.sourceError) {
+      console.log('Original Error', e.error?.details?.sourceError);
+    }
+  }
+
+  /**
+   * Decides whether to display video or play audio based on the track type.
+   * @param {Object} event - The track event object.
+   */
+  displayTrack(event) {
+    if (event.track.kind === 'video') {
+      this.displayVideo(event);
+    } else {
+      this.playAudio(event);
+    }
+  }
+
+  /**
+   * Adds a <video> element to the DOM and attaches the video track to it.
+   * @param {Object} event - The track event object containing the video track.
+   */
+  displayVideo(event) {
```

Comment:
This looks to create a new video div each time a track starts - what happens if the track changes when a track was already started before, issuing a new `"track-started"` event for the same participant? (e.g., in case of device change or meeting move)

Reply:
The `videos` div is constant. I'm creating a new `<video>` element which I append to the `videos` div. In testing it, it does what I would expect. I think this is OK, but let me know if you have any concerns.

Reply:
Ah sorry I misspoke when I said the div. I did mean the video element. Basically, what happens if a participant who already _has_ a video element here in this div gets _another_ `"track-started"` event for any reason? (I'm not sure if you permit device switching here but specifically a meeting move comes to mind, which has bitten me in demos when I don't account for it before!). It seems like we will end up appending _another_ video element for the same participant without destroying the old one, no?